lib/prerequisite.js

```diff
@@ -9,6 +9,60 @@ module.exports = (input, pkg, opts) => {
 
 	const tasks = [
 		{
+			title: 'Ping npm registry',
+			task: () => new Promise((resolve, reject) => {
+				// `npm ping` waits wayyyy too long
+				const timeout = setTimeout(() => {
+					reject(new Error('Connection to npm registry timed out'));
+				}, 6000);
+				execa.stdout('npm', ['ping'])
+					.then(() => {
+						clearTimeout(timeout);
+						resolve();
+					}, () => {
+						throw new Error('Connection to npm registry failed');
+					});
+			})
+		},
+		{
+			title: 'Verify user authenticated',
+			task: () => new Promise((resolve, reject) => {
```

Comment:
The below can be done without a Promise constructor. Now you're just wrapping promises in a new promise.